recursion.rb

```diff
@@ -1,14 +1,48 @@
-
+#Recurisive cas: n! = n *(n-1)!
+#Basecase: 1! = 1
+#Basecase: 0! = 1
 def fact(n)
+  if n == 1 || n == 0
+    return 1
+  end
+  return n * fact(n-1)
 end
 
+# fib(1) = 1
+# fib(2) = 1
+# fib(3) = 2
+# fib(3) = fib(2) + fib(1)
+# fib(4) = fib(3) + fib(2)
 def fib(n)
+  if n == 1 || n == 2
+    return 1
+  end
+  return fib(n-1) + fib(n-2)
 end
 
 def pal(s)
+  if s.length == 1 || s.length == 0
+    return true
+  end
+  if s[0] == s[-1]
+     pal(s[1..-2])
+  else
+    return false
+  end
 end
 
 def binary(n)
+  binary_helper(n,"","")
+end
+
+def binary_helper(digits, binary, result)
+  if digits == 0
+    result += binary + " "
+  else
+  result = binary_helper(digits-1,binary+"0", result)
+  result = binary_helper(digits-1,binary+"1", result)
+  end
+  return result
 end
```

Comment:
Thanks so much for taking the time to do this review! Much appreciated! Suzanne

Sent from my iPhone

> On May 15, 2016, at 9:43 PM, malderi <notifications@github.com> wrote:
> 
> In recursion.rb:
> 
> >  end
> >  
> >  def binary(n)
> > +  binary_helper(n,"","")
> > +end
> > +
> > +def binary_helper(digits, binary, result)
> > +  if digits == 0
> > +    result += binary + " "
> > +  else
> > +  result = binary_helper(digits-1,binary+"0", result)
> > +  result = binary_helper(digits-1,binary+"1", result)
> > +  end
> > +  return result
> >  end
> Good solution.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>